src/family_name_classifier.py

```diff
@@ -0,0 +1,315 @@
+import builtins
+
+import pandas as pd
+import numpy as np
+import itertools
+import unicodedata
+import re
+import collections
+import random
+import sklearn.feature_extraction
+import sklearn.linear_model
+import sklearn.cross_validation
+import sklearn.externals.joblib
+random.seed(0)
+
+
+class FamilyNameClassifier:
+    IGNORE_LIST = {'de', 'da', 'do', 'des', 'das', 'dos', 'e', 'dr', 'sobrinho', 'junior', 'neto', 'jr', 'filho'}
+    NOT_ACCEPTABLE_CHARACTERS = re.compile(r'[^a-z ]')
+
+    def __init__(self):
+        self.vectorizer = None
+        self.surname_counter = None
+        self.total_surnames = None
+        self.total_first_names = None
+        self.model = None
+
+    def tokenize(self, name):
+        """"
+        Simple tokenizer that normalizes and split a name by spaces.
+
+        :param name: string
+        :return: tokens: list
+        """
+
+        name = name.strip().lower()
+        name = unicodedata.normalize('NFD', name)
+        name = self.NOT_ACCEPTABLE_CHARACTERS.sub("", name)
```

Comment:
Maybe if the script don't work with special chars we could some of them in ASCII characters:

```python
import unicodedata

value = 'fóãçêé'

for c in unicodedata.normalize('NFD', value):
    if unicodedata.category(c) != 'Mn':
        print(c)
```

Outputs:

```console
f
o
a
c
e
e
```